theano/gof/vm.py

```diff
@@ -152,7 +152,11 @@ def update_profile(self, profile):
             profile.node_cleared_order = self.node_cleared_order[:]
 
         if hasattr(self, 'dependencies'):
-            profile.dependencies = self.dependencies.copy()
+            if self.dependencies:
+                profile.dependencies = self.dependencies.copy()
+            else:
+                profile.dependencies = {}
+
```

Comment:
Why this case can happen? I think the old code was good. I think self.dependencies is always a Dict. When this isn't exact?